torchmetrics/metric.py

```diff
@@ -363,10 +439,15 @@ def _load_from_state_dict(
         error_msgs: List[str],
     ) -> None:
         """ Loads metric states from state_dict """
+
+        # only global rank 0 should be reloading the values present in the ``state_dict``
+        # as the state contains synced values across all progress_group
         for key in self._defaults:
             name = prefix + key
             if name in state_dict:
-                setattr(self, key, state_dict.pop(name))
+                value = state_dict.pop(name)
```

Comment:
This will modify the passed state dict, do we want that?

Reply:
Yes, as it would fail otherwise with unexpected parameters.

Reply:
maybe copy the state_dict first then?

Otherwise, the user will be left out with a different `state_dict` than the one he passed

Reply:
This behaviour was already present in nn.Metric and should be changed in another PR. But I would argue this seems to be a common way to do.